### PR TITLE
Stop on error during packaging

### PR DESCRIPTION
**Pull request by sruke**
Throw an error if there was an error during packaging.

### CODE REVIEW

File: pack.ps1 [hunk 1]
```diff
@@ -63,11 +63,21 @@ Write-Host "releaseVersion: " $releaseVersion;
...
 CreateArtifactsFolder($artifactsRoot);
-foreach($project in $buildConfiguration.SelectNodes("root/projects/src/project"))
+
+foreach ($project in $buildConfiguration.SelectNodes("root/projects/src/project"))
 {
-	$name = $project.name;
-	Write-Host ">>> Start-Process -wait -NoNewWindow $dotnetexe 'pack' --no-build --no-restore -nodereuse:false -c $buildType -o $artifactsRoot -v q -s $root\src\$name\$name.csproj"
-	Start-Process -Wait -PassThru -NoNewWindow $dotnetexe "pack --no-build --no-restore -nodereuse:false -c $buildType -o $artifactsRoot -v q -s $root\src\$name\$name.csproj"
+    $name = $project.name
+    $projectPath = "$root\src\$name\$name.csproj"
```
Comment (keegan-caruso) [inline pack.ps1]: this won't work on linux/mac right?

How about [System.IO.Path]::Combine instead?

File: src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
+<Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Suppression>
+    <DiagnosticId>CP0002</DiagnosticId>
+    <Target>M:Microsoft.IdentityModel.Tokens.JsonWebKeyConverter.ConvertFromECDsaSecurityKey(Microsoft.IdentityModel.Tokens.ECDsaSecurityKey)</Target>
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml:6]: oh I think I see what the tool is trying to catch.

net461 implements netstandard 2.0

So this is a bit confusing. Is there any reason ConvertFromECDsaSecurityKey cannot also be included in net461?

File: pack.ps1 (hunk 1 above)
Comment (sruke) [reply pack.ps1]: My understanding is this script is not used by Wilson Linux builds.

File: src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml (hunk 2 above)
Comment (sruke) [reply src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml:6]: I'm not aware of the history behind this decision. A [GitHub Issue](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2363) has been opened to fail gracefully if a API attempts to invoke this method on net461.

Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml:6]: @keegan-caruso even though the api is available in netstandard 2.0, we found 461 and 462 did not support this api.
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.eccurve?view=netstandard-2.0
Some types are not supported.

Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml:6]: interesting!

Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/CompatibilitySuppressions.xml:6]: @keegan-caruso i wrote 472 above, but i meant 462.
What puzzles me is why only an error for 461 and not 462?

**Review by brentschmaltz — APPROVED** [review]

File: pack.ps1 (hunk 1 above)
Comment (sruke) [reply pack.ps1]: Updated to use Path.Combine.

**Review by keegan-caruso — APPROVED** [review]

**Pull request merged**